agent/consul/state/config_entry.go

```diff
@@ -392,7 +377,6 @@ var serviceGraphKinds = []string{
 
 func (s *Store) validateProposedConfigEntryInServiceGraph(
```

Comment:
This isn't related to your changes but I just noticed that in L415 we check for the err from `tx.Get(configTableName, "link", sid)` after iterating on `iter`. 

I think the err check should be moved up.